pulsar-client/src/main/java/org/apache/pulsar/client/impl/PersistentAcknowledgmentsGroupingTracker.java

```diff
@@ -560,7 +529,6 @@ private void flushAsync(ClientCnx cnx) {
     @Override
     public void flushAndClean() {
         flush();
-        lastCumulativeAck = LastCumulativeAck.create((MessageIdImpl) MessageIdImpl.earliest, null);
```

Comment:
if delete this line when seek the message to early position, the consumer will ack the incorrect position

Reply:
The original code uses `cumulativeAckFlushRequired` to determine whether to send ACK in `flushAsync`. Even if `lastCumulativeAck` is reset to earliest here, the `lastCumulativeAck` won't work without calling `doCumulativeAckAsync`, which is the only place that changes `cumulativeAckFlushRequired` with true. And after `doCumulativeAckAsync`, the internal message ID of `lastCumulativeAck` will change, so I think this line here is useless. Could you show a test to verify your point?

In this PR, after `lastCumulativeAck.flush()` is called, the internal `flushRequired` field became false and `lastCumulativeAck` wouldn't work in `flushAsync`.

Reply:
I added the comparison log with `messageId` in `LastCumulativeAck#update` back. So we need to reset the `lastCumulativeAck` now to allow it being updated with any value.